src/html/Quoting.cc

```diff
@@ -9,50 +9,48 @@
 #include "squid.h"
 #include "html/Quoting.h"
 
-/*
- *  HTML defines these characters as special entities that should be quoted.
- */
-static struct {
-    unsigned char code;
-    const char *quote;
-} htmlstandardentities[] =
+#include <array>
+#include <cstring>
+
 
+static const auto& MakeEscapeSequences()
 {
-    /* NOTE: The quoted form MUST not be larger than 6 character.
-     * see close to the MemPool commend below
+    static std::array<std::pair<unsigned char, const char *>, 5> const escapePairs = {
+        std::make_pair('<', "&lt;"),
+        std::make_pair('>', "&gt;"),
+        std::make_pair('"', "&quot;"),
+        std::make_pair('&', "&amp;"),
+        std::make_pair('\'', "&apos;")
+    };
+    static auto escapeMap = new std::array<const char *, 256>{};
```

Comment:
As we discussed on the conference call, I do recommend migrating this array from c-string to SBuf in this PR after removing tools/cachemgr.cgi, but it is your call.

Reply:
Waiting for PR #1542 to land

Reply:
done